composables/useAuth.ts

```diff
@@ -0,0 +1,267 @@
+import auth0, { Auth0UserProfile } from "auth0-js";
+import Cookies from "js-cookie";
+
+const userData = ref<Auth0UserProfile | null>(null);
+const auth = ref<auth0.WebAuth | null>(null);
+
+const useAuth = () => {
+	const config = useRuntimeConfig();
+	const AUTH_DOMAIN = config.public.auth0.AUTH_DOMAIN; // DOMAUN from https://auth0.com/ application
+	const AUTH_CLIENTID = config.public.auth0.AUTH_CLIENTID; // CLIENTID from https://auth0.com/ application
+
+	const isLoggedIn = computed(() => !!userData.value);
+
+	if (!auth.value) {
+		if (!AUTH_DOMAIN || AUTH_CLIENTID) {
+			console.error(
+				"runtimeConfig.public.auth0.AUTH_DOMAIN & runtimeConfig.public.auth0.AUTH_DOMAIN are required to be set in nuxt.config, for useAuth to work"
```

Comment:
vill sætte den helt i starten for at fjerne nested if statemenst samt slet ikke starte noget hvis env ikke eksistrere